bigchaindb/models.py

```diff
@@ -74,29 +82,26 @@ def validate(self, bigchain):
                     raise DoubleSpend('input `{}` was already spent'
                                       .format(input_txid))
 
-                input_conditions.append(input_tx.conditions[input_cid])
+                input_condition = input_tx.conditions[input_cid]
+                input_conditions.append(input_condition)
+
                 input_txs.append(input_tx)
-                if input_tx.conditions[input_cid].amount < 1:
-                    raise AmountError('`amount` needs to be greater than zero')
-                input_amount += input_tx.conditions[input_cid].amount
 
             # validate asset id
-            asset_id = Asset.get_asset_id(input_txs)
-            if asset_id != self.asset.id:
+            asset_id = Transaction.get_asset_id(input_txs)
+            if asset_id != self.asset['id']:
                 raise AssetIdMismatch(('The asset id of the input does not'
                                        ' match the asset id of the'
                                        ' transaction'))
 
-            # get the asset creation
-            asset = bigchain.get_asset_by_id(asset_id)
-            # validate the asset
-            asset.validate_asset(amount=input_amount)
             # validate the amounts
-            output_amount = 0
-            for condition in self.conditions:
-                if condition.amount < 1:
-                    raise AmountError('`amount` needs to be greater than zero')
-                output_amount += condition.amount
+            if any(any(condition.amount < 1 for condition in conditions) for
```

Comment:
> You already validate the amounts

Ah, but that validation was only for `CREATE` and `GENESIS` transactions.

> And the input conditions where already validated when the input transaction was submitted

Hmmm, actually that's a good point... the input conditions *must* be valid since we get them from valid blocks.